primitives/primproc/batchprimitiveprocessor.cpp

```diff
@@ -1467,6 +1471,19 @@ void BatchPrimitiveProcessor::execute()
           asyncLoaded[p] = true;
         }
 
+        if (col->hasAuxCol())
```

Comment:
I am curios why did we add this loadBLockAsync if filterCount == 1 && hasScan? I got the impression that aux column must be loaded in any case.

Reply:
AUX column is tightly coupled with the original scan column. So I am guarding it under the same condition under which we perform `loadBlockAsync` on the scan column, which is `if (filterCount == 1 && hasScan)`. There is a comment at the start of this `if` block: "`// if only one scan step which has no predicate, async load all columns`". I am not fully sure the reasoning behind this comment.